core/services/vrf/integration_v2_test.go

```diff
@@ -260,88 +277,152 @@ func TestIntegrationVRFV2_OffchainSimulation(t *testing.T) {
 	// job is fully up and running and not add it to the active jobs list before then)
 	time.Sleep(2 * time.Second)
 
-	subFunding := decimal.RequireFromString("1000000000000000000")
-	_, err = uni.consumerContract.TestCreateSubscriptionAndFund(uni.carol,
-		subFunding.BigInt())
+	return
+}
+
+func TestIntegrationVRFV2_OffchainSimulation(t *testing.T) {
+	config, _, _ := heavyweight.FullTestDB(t, "vrf_v2_integration_sim", true, true)
+	ownerKey := cltest.MustGenerateRandomKey(t)
+	uni := newVRFCoordinatorV2Universe(t, ownerKey, 3)
+	app := cltest.NewApplicationWithConfigAndKeyOnSimulatedBlockchain(t, config, uni.backend, ownerKey)
+	config.Overrides.GlobalEvmGasLimitDefault = null.NewInt(0, false)
+	config.Overrides.GlobalMinIncomingConfirmations = null.IntFrom(2)
+
+	// subscribe the consumers to vrf
+	subs := []vrf_coordinator_v2.GetSubscription{}
+	for i := 0; i < len(uni.vrfConsumers); i++ {
+		subs = append(subs,
+			subscribeVRF(t, uni.vrfConsumers[i], uni.consumerContracts[i], uni.rootContract, uni.backend))
+	}
+
+	carol := uni.vrfConsumers[0]
+	carolContract := uni.consumerContracts[0]
+
+	// Lets create 2 gas lanes
+	key1, err := app.KeyStore.Eth().Create(big.NewInt(1337))
 	require.NoError(t, err)
-	uni.backend.Commit()
-	sub, err := uni.rootContract.GetSubscription(nil, uint64(1))
+	sendEth(t, ownerKey, uni.backend, key1.Address.Address(), 10)
+	key2, err := app.KeyStore.Eth().Create(big.NewInt(1337))
 	require.NoError(t, err)
+	sendEth(t, ownerKey, uni.backend, key2.Address.Address(), 10)
+
+	gasPrice := decimal.NewFromBigInt(big.NewInt(10e9), 0) // Default is 10 gwei
+	configureSimChain(app, map[string]types.ChainCfg{
+		key1.Address.String(): {
+			EvmMaxGasPriceWei: utils.NewBig(big.NewInt(10e9)), // 10 gwei
+		},
+		key2.Address.String(): {
+			EvmMaxGasPriceWei: utils.NewBig(big.NewInt(100e9)), // 100 gwei
+		},
+	}, gasPrice.BigInt())
+	require.NoError(t, app.Start())
+
+	jbs := createJobs(t, []ethkey.KeyV2{key1, key2}, app, uni)
+
+	// enqueue requests for the other consumers to test that their balance doesn't affect carol's
+	// we're not really going to do anything about these requests, so they'll sit around for a bit
+	// for the duration of the test.
```

Comment:
the other consumer txes do actually go through though, so maybe update this comment